rope/base/pynamesdef.py

```diff
@@ -24,7 +24,10 @@ def get_object(self):
     def get_definition_location(self):
         """Returns a (module, lineno) tuple"""
         if self.lineno is None and self.assignments:
-            self.lineno = self.assignments[0].get_lineno()
+            try:
+                self.lineno = self.assignments[0].get_lineno()
+            except AttributeError:
```

Comment:
I am really hesitant to let this go in without real understanding how it may happen that `self.assignments[0]` doesn’t have `get_lineno()` child. Where are all places where this property can be filled from?

Reply:
Yeah, I agree with you that's not ideal. I'm not incredible familiary with `rope`, I tried to wonder a bit in the codebase but it's reallt interleaved and there are several layers of abstractions and I'm having an hard time finding the culprit. 

Apparently, in my project, the `ast_node` is `None` when this exception is triggered.

Reply:
The problem is I'm not able to find a MWE to reproduce the problem.

Reply:
The problem is that I don’t use ropevim anymore myself (see https://github.com/mcepl/rope-language-server). Do you want to take over ropevim?

Reply:
Oh! Thanks, I was looking for something like `rope-language-server`. Is that usable with `vim`? Maybe via `YouCompleteMe`? I'll be happy to switch in that case...

Reply:
(Thanks for offering to take over ropevim, but I'm not at all proficient with it and even less with language parsing, lexing, etc... Not sure I'll be the right person for the job. Thanks anyway!)

Reply:
(Also, not incredible proficient with Vimscript either)

Reply:
`rope-language-server` is just [LSP](https://microsoft.github.io/language-server-protocol/implementors/servers/) and you can use it with any LSP client (neovim-0.5, and [others](https://microsoft.github.io/language-server-protocol/implementors/tools/).